### ISSUE TITLE
Running example.cpp causes segmentation fault after PR #13

### ISSUE THREAD

**Issue by HugoKooijman** [dmikushin/binance-cxx-api#14]
Hello,

Since I updated the changes after #13, I get a segmentation fault when running example.cpp. I narrowed down the cause to line 78: Websocket::enter_event_loop();
Had 0 problems before, so I'm positive it's since last update. Sometimes it happens instantly, sometimes it takes longer (10-20 seconds). So I'm guessing it happens in the do-while loop.

OS info:
      System Version: macOS 10.14.6 (18G103)
      Kernel Version: Darwin 18.7.0

Thank you for your time

**Comment by HugoKooijman** [comment]
Apologies, it appears it was a compiler issue. I'll be more careful with opening issues too hastily in the future.

**Comment by dmikushin** [comment]
Hi @HugoKooijman No worries, I've just recompiled from scratch on Catalina and ran example for a while without any issue.

**Comment by tjappo** [comment]
@dmikushin Hey, while fetching a single candle still works, I am also experiencing a segmentation fault when trying to fetch using WebSockets after printing the klines cache. It gives the segmentation fault when the json candle data is received.

**Comment by dmikushin** [comment]
@tjappo Could you please give us a test case?

**Comment by tjappo** [comment]
> @tjappo Could you please give us a test case?

I am just running the example.

**Comment by dmikushin** [comment]
We ship our own jsoncpp as a submodule and use it, in order to avoid possible issue. So you should be fine on that side. The backtrace looks a lot like an unexpected NULL pointer is involved. Without being able to reproduce the issue, I've added an extra check here: afe3acad2bb159d738bd613c2a32cff6c073e419 This may help or may not, anyway it would be much better to have debug information. This is how:

```
cmake -DCMAKE_BUILD_TYPE=Debug ..
make -j12
lldb ./example
(lldb) run
```

Recompile as shown and run in lldb. Then you should be brought to the crash site with the source code. Please post us the lldb log, thanks!

**Comment by tjappo** [comment]
It was due to the usage of the newest version of the jsoncpp [00.11.0](https://github.com/open-source-parsers/jsoncpp/releases/tag/00.11.0). I'll come back to you with the lldb log.

**Comment by tjappo** [comment]
@dmikushin Thanks for the fix [afee3aca](https://github.com/dmikushin/binance-cxx-api/commit/afe3acad2bb159d738bd613c2a32cff6c073e419), it worked with the newest version of jsoncpp! If you can merge it into the master, you can also push the version of the jsoncpp, or should I make a pr to update it?

**Comment by dmikushin** [comment]
You are right, we are running into conflict with system's jsoncpp, because our jsoncpp headers inclusion is missing for example target. You should see it fixed after updating to d880a88c5a54a03c8cc171150314f8f23ea5eaca Thanks for spotting this!
